Sources/AudioKitUI/Visualizations/SpectrogramFlatView/SpectrogramModel.swift

```diff
@@ -0,0 +1,173 @@
+// Copyright AudioKit. All Rights Reserved. Revision History at http://github.com/AudioKit/AudioKitUI/
+//
+
+import AudioKit
+import SwiftUI
+
+struct SpectrogramFFTMetaData {
+    // fftSize defines how detailled the music is analyzed in the time domain. 
+    // the lower the value, the less detail:
+    // * 1024: will receive about four analyzed frequencies between C2 and C* (65Hz to 130Hz). 
+    //             New data comes roughly 21.5 times per second, each 46ms.  
+    // * 2048: will receive about eight analyzed frequencies between C2 and C* (65Hz to 130Hz).  
+    //            New data comes roughly 11 times per second, each 93ms. 
+    // * 4096: will receive about 16 analyzed frequencies between C2 and C* (65Hz to 130Hz). 
+    //            New data comes roughly  5.5 times per second, each 186ms.  
+    // Choose a higher value when you want to analyze low frequencies, 
+    // choose a lower value when you want fast response and high frame rate on display.
+    // Default: 2048
+    var fftSize = 4096/2
+
+    // how/why can the sample rate be edited? Shouldn't this come from the node/engine?
+    // if the sample rate is changed, does the displayed frequency range also have to be changed?
+    // took this from existing SpectrogramView, will investigate later
+    var sampleRate: double_t = 44100
+}
+
+@available(iOS 17.0, *)
+struct SliceQueue {
+    var maxItems: Int = 120
+    var items: [SpectrogramSlice] = []
+
+    public mutating func pushToQueue(element: SpectrogramSlice) {
+        enqueue(element: element)
+        if items.count > maxItems {
+            dequeue()
+        }
+    }
+
+    private mutating func enqueue(element: SpectrogramSlice) {
+        items.append(element)
+    }
+
+    private mutating func dequeue() {
+        if !items.isEmpty {
+            items.remove(at: 0)
+        }
+    }
+}
+
+///  Model for the SpectrogramFlatView. Makes connection to the audio node and receives FFT data
+@available(iOS 17.0, *)
+class SpectrogramFlatModel: ObservableObject {
```

Comment:
if this code does indeed require iOS 17, then no reason to use newly deprecated `ObservableObject` and should use `@Observable` macro